apollo-backend/src/main/java/io/logz/apollo/services/BlockerService.java

```diff
@@ -188,4 +205,18 @@ private boolean isBlockerInScope(Blocker blocker, Deployment deployment) {
 
         return false;
     }
+
+    public synchronized BlockerFunctionResponse checkDeploymentShouldBeBlockedByServiceByRegionBlocker(List<Integer> serviceIds, List<Integer> environmentIds) {
```

Comment:
synchronized doesn't really solve the problem
if 2 users deploy in parallel, they will execute this function one after another, but you still may end up allowing both if the 2nd request read from the database before the 1st one add deployment to the database in `io.logz.apollo.deployment.DeploymentHandler#addDeployment`

read and update operation should be atomic

additionally, the scope of a synchronized is too big.
I suggest you to find the smallest possible scope inside `SingleRegionBlocker.shouldBlock` function